enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,396 @@
+3. Enable continuous monitoring of tags applied and reconcile when there are changes made external to 
+the controller's API.
+4. Provide CLI for operations, cloud service provide authentication and ability to edit custom resource with ease.
+
+### Non-Goals
+
+1. Managing metadata which are not related to cloud resource tagging.
+2. Sub-set of input tags to be applied on selected cloud resource.
+3. Any conflict with cloud service provider tagging policies will not be managed by controller.
+4. Regular expression or wildcard characters for classifier tag.
+5. Metrics collection and audit logging features for controller and user operations.
+6. UI frontend for managing tags.
+
+## Proposal 
+
+A new kubernetes controller to generates actions based on tag specification defined by a custom resource. The controller
+reads existing data, applies tags based on opcode and maximum limit policy. The controller would require an initial tag or label, 
+that can be used to identify cloud resources to apply and manage tags. Cloud resources can be ignored from being managed 
+by the controller by removing the classifier tag on the cloud resource. Controller reconciles the tag list on 
+cloud resources periodically at intervals configured by sync period for the controller. 
+
+### Controller configuration
+
+#### Maximum limit policy
+
+Maximum limit policy indicates whether the input tag list can be applied partially when the length of the list exceeds the 
+allowed limit on the cloud resource. The result might differ when applied along overwrite policy. Following are the list of valid values.
+
+1. POLICY_APPLY_PARTIAL - Allows input tag list applied to be partial on cloud resource. 
+2. POLICY_APPLY_STRICT - Applies input tag list strictly in complete.
+
+### Opcodes
+
+Controller supports different opcodes for actions to add, update and delete tags. Following are the supported opcodes
+1. "add" - Controller adds tags to the cloud resources. In case of an existing entry for the tag, controller reports failure.
+2. "update" - Controller updates existing tags for the cloud resources. If there is no existing entry, controller reports failure.
+3. "delete" - Controller removes tags from the cloud resources. In case of specific errors, user intervention may be required to remove tags manually
+from the cloud resources.
+
+### Workflow Description
+
+#### Bootstrap
+1. User starts the controller with cloud service provider credentials and sync period set.
+2. Controller starts to list/watch custom resource.
+3. User creates `AWSMetadata` object with tag list details.
+4. User creates `CloudMetadata` configuration with controller configuration.
+5. Based on the configuration, controller queries and lists all resources based on classifier tag.
+6. Creates list of existing tags on cloud resource.
+7. Add/update tag workflow is initiated.
+
+#### Add/Update tags
+
+1. User creates `AWSMetadata` object with tag list details.
```

Comment:
Wonder, have you considered the idea of having each tag represented as a single CR, the tag lifecycle then being tied to just a single CR instance, then the removal of the CR from the Kube cluster would represent either the removal from the cloud or orphaning of the tag based on some removal policy, might be cleaner than a singleton CR that houses a number of tags